src/main/java/net/mcreator/ui/browser/WorkspaceFileBrowserContextMenu.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2021, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.browser;
+
+import net.mcreator.generator.GeneratorFlavor;
+import net.mcreator.ui.component.tree.FilterTreeNode;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.init.UIRES;
+
+import javax.swing.*;
+import java.io.File;
+
+class WorkspaceFileBrowserContextMenu extends JPopupMenu {
+
+	WorkspaceFileBrowserContextMenu(WorkspaceFileBrowser browser) {
+		boolean fileActionsAllowed = false;
+		JMenu createMenu = L10N.menu("workspace_file_browser.add");
+		createMenu.setIcon(UIRES.get("16px.add.gif"));
+
+		FilterTreeNode selected = (FilterTreeNode) browser.tree.getLastSelectedPathComponent();
+		if (selected.getUserObject() instanceof File file) {
+			if (file.isFile())
+				file = file.getParentFile();
+
+			if (browser.mcreator.getGeneratorConfiguration().getGeneratorFlavor().getBaseLanguage()
+					== GeneratorFlavor.BaseLanguage.JAVA) {
+				if (file.isDirectory() && file.getAbsolutePath()

```

Comment:
You should still use canonical path as asked, just don't wrap the whole thing in try, only parts needed, maybe make new helper method in FileIO "is file A inside B"

Reply:
So that was a good idea

Reply:
:heavy_check_mark: Fixed

Reply:
Request review when PR is fully ready :)

Reply:
...
It is

Reply:
You should indicate this by requesting a review, as GitHub workflow goes